compiler/damlc/daml-doc/src/DA/Daml/Doc/HaddockParse.hs

```diff
@@ -67,16 +67,23 @@ mkDocs opts fp = do
                     case stripInstanceSuffix (cl_name classDoc) of
                         Nothing -> Left classDoc
                         Just templateName -> Right (templateName, classDoc)
-            templateInstanceMap = MS.fromList templateInstanceClasses
+            templateInstanceClassMap = MS.fromList templateInstanceClasses
 
             md_name = dc_modname
             md_anchor = Just (moduleAnchor md_name)
             md_descr = modDoc dc_tcmod
-            md_templates = getTemplateDocs ctx typeMap templateInstanceMap
+            md_templates = getTemplateDocs ctx typeMap templateInstanceClassMap
             md_functions = mapMaybe (getFctDocs ctx) dc_decls
-            md_adts
+
+            filteredAdts -- all ADT docs without templates or choices
                 = MS.elems . MS.withoutKeys typeMap . Set.unions
-                $ dc_templates : MS.keysSet dc_templateInstances : MS.elems dc_choices
+                $ dc_templates : MS.elems dc_choices
+
+            (md_adts, md_templateInstances) =
```

Comment:
`md_adts` does not contain all ADTs right? It looks like it only contains template and choice ADTs that are not template instances?